### PR TITLE
fix(discover): Non Team projects weren't saveable

### PR DESCRIPTION
**Pull request by wmak**
- If a user wasn't in a team they couldn't save discover queries with
  projects that still belong to the organizaion

### CODE REVIEW

File: src/sentry/discover/endpoints/serializers.py [hunk 1]
```diff
@@ -177,7 +177,7 @@ def validate_projects(self, projects):
             return projects
 
         # Params will have project_ids that have been validated
-        if projects != set(self.context["params"]["project_id"]):
+        if not projects.intersection(set(self.context["params"]["project_id"])):
```
Comment (markstory) [inline src/sentry/discover/endpoints/serializers.py:180]: Will this still fail if projects from outside the organization are requested to be added to the query?

Comment (wmak) [reply src/sentry/discover/endpoints/serializers.py:180]: Yep `test_save_with_wrong_projects` checks this, in the case the query is `project:some_project_from_another_org`

`projects` here would still have the other org's project but
`self.context["params"]["project_id"]` will only have projects for the user's org

Comment (markstory) [reply src/sentry/discover/endpoints/serializers.py:180]: Yes the half valid, half invalid scenario is the one I was thinking of.

Comment (wmak) [reply src/sentry/discover/endpoints/serializers.py]: 🙏 Thanks, updated to avoid allowing that case

**Review by markstory — APPROVED** [review]

**Review by evanh — APPROVED** [review]

**Pull request merged**